Adding Non distilled Whisper

stisiTT opened the pull request:
This PR adds Non-Distilled Whisper model to the run.py workflows.

stisiTT committed:
Non-Distilled Whisper Weights Swap

stisiTT committed:
Non-Distilled Whisper Weights Swap plus Distilled

Re adding Distilled models

tstescoTT commented on setup.sh:
```diff
@@ -9,6 +9,7 @@ set -euo pipefail  # Exit on error, print commands, unset variables treated as e
 usage() {
     echo "Usage: $0 <model_type>"
     echo "Available model types:"
+    echo "  Whisper-Large-v3"
```
setup.sh is deprecated, can this use `workflows/setup_host.py`?

stisiTT committed:
Updating to reset tt-metal and vllm builds

stisiTT committed:
Dockerfile Update

stisiTT committed:
Add whisper model type configuration and update docker environment variables

- Introduced `whisper_model_type` attribute in `ModelConfig` to specify model type ("original" or "distilled").
- Updated `config_list` to set `whisper_model_type` for both original and distilled models.
- Updated `run_docker_server` to include `WHISPER_MODEL_TYPE` in the Docker environment variables if applicable.

stisiTT commented on evals/eval_config.py:
```diff
@@ -727,6 +727,41 @@ def validate_data(self):
+                    # average score over LibriSpeech clean & other
+                    # score is Word-Error-Rate, so turn it into
+                    # "Word-Success-Rate"
+                    published_score=(100 - 5.25),
+                    gpu_reference_score=(100 - 3.805),
```
@gtobarTT are you able to run these?

stisiTT commented on evals/eval_config.py:
```diff
@@ -727,6 +727,41 @@ def validate_data(self):
     # TODO: Probably create DockerEvalConfig because this doesn't make sense to
     # mix with these "vLLM" eval configs
+    EvalConfig(
+        hf_model_repo="openai/whisper-large-v3",
```
Taken care of https://github.com/tenstorrent/tt-metal/commit/87a1921de12993e6b789bb8a28e5ae621df3f878

stisiTT committed:
Commit to tt-metal SHA update

bgoelTT reviewed (commented):
Thanks Sam looks good, can you please run the eval to see what score we get on TT?

stisiTT commented:
> Thanks Sam looks good, can you please run the eval to see what score we get on TT?

@bgoelTT it's currently running. I've been asked to make the tt-metal change to be a pytest arg instead of an environment variable, so I'm waiting for that work to wrap up, but I can still get numbers running on TT

stisiTT committed:
Refactor whisper model configuration and update Docker environment variables

- Changed `whisper_model_type` to `whisper_model_repo` in `ModelConfig` to directly reference the HF repo ID for whisper models.
- Updated `config_list` to set `whisper_model_repo` for both original and distilled models.
- Modified `run_docker_server` to use `WHISPER_MODEL_REPO` instead of `WHISPER_MODEL_TYPE` in the Docker environment variables.

stisiTT committed:
Updating Model Config to point to reset branch commit

stisiTT committed:
Commit to tt-metal change

stisiTT committed:
Updating tt-metal commit to main

stisiTT committed:
uncommenting other ModelConfigs

stisiTT committed:
Fixing out of place updates

Updates to Qwen model versions which shouldn't be a part of this branch

stisiTT commented:
The last commit to tt-metal, which should have just been formatting changes with pre-commit, are breaking the evals

stisiTT committed:
Enhance Docker evaluation command for whisper models

- Updated `build_docker_eval_command` to select the appropriate model repository based on the presence of `whisper_model_repo`.
- Added environment variable `WHISPER_MODEL_REPO` in the whisper evaluation script to ensure correct model configuration during execution.

stisiTT committed:
Updating lmms-eval branch

stisiTT committed:
Add LibriSpeech evaluation scope support

- Introduced `--librispeech-scope` argument in `run.py` and `run_docker_evals.py` to specify evaluation scope for LibriSpeech tasks.
- Implemented functions in `eval_config.py` to generate result keys, task names, and evaluation scores based on the selected scope.
- Updated Docker evaluation workflow to dynamically configure LibriSpeech tasks based on the provided scope.

stisiTT commented:
| Task Name              | Metric   |  Value |
|------------------------|----------|--------|
| librispeech_dev_clean  | wer,none | 2.6005 |
| librispeech_dev_other  | wer,none | 5.6376 |
| librispeech_test_clean | wer,none | 2.5515 |
| librispeech_test_other | wer,none | 5.6106 |


### Accuracy Evaluations for distil-large-v3 on n300

| model           | device | task_name   | accuracy_check | score | ratio_to_reference | gpu_reference_score | ratio_to_published | published_score                           |
|-----------------|--------|-------------|----------------|-------|--------------------|---------------------|--------------------|-------------------------------------------|
| distil-large-v3 | n300   | librispeech | PASS ✅         | 95.90 | 1.00               | 96.19               | 1.01               | [94.75](https://arxiv.org/pdf/2311.00430) |




openai/whisper-large-v3
|        Tasks         |Version|Filter|n-shot|Metric|   |Value |   |Stderr|
|----------------------|-------|------|-----:|------|---|-----:|---|------|
|librispeech_test_other|Yaml   |none  |     0|wer   |↓  |4.4193|±  |   N/A|

stisiTT committed:
Refactor LibriSpeech evaluation functions and update Docker task configuration

- Renamed and refactored `get_librispeech_result_keys` to `_get_whisper_librispeech_result_keys` for clarity and encapsulation.
- Removed redundant LibriSpeech scope handling from `run_docker_evals.py` and integrated dynamic task configuration based on the evaluation scope.
- Updated evaluation tasks to utilize the new result keys and score configurations based on the specified scope.

stisiTT committed:
Refactor audio evaluation dataset handling and update configurations

- Renamed `--librispeech-scope` argument to `--audio-eval-dataset` in `run.py` and `run_docker_evals.py` for clarity.
- Updated choices and default values for the audio evaluation dataset argument to include OpenSLR and LibriSpeech subsets.
- Refactored evaluation functions in `eval_config.py` to accommodate the new dataset argument and ensure backward compatibility with legacy scope handling.
- Adjusted Docker evaluation workflow to dynamically configure tasks based on the selected audio evaluation dataset.

stisiTT committed:
Refactor audio evaluation dataset integration and enhance configuration handling

- Introduced `AUDIO_EVAL_DATASETS` in `eval_config.py` to centralize audio evaluation dataset options.
- Updated `--audio-eval-dataset` argument in `run.py` and `run_docker_evals.py` to utilize the new centralized dataset choices.
- Refactored evaluation logic in `run_docker_evals.py` to dynamically configure tasks based on the selected audio evaluation dataset, improving clarity and maintainability.

stisiTT committed:
Update default audio evaluation dataset and enhance configuration options

- Changed the default value of the `--audio-eval-dataset` argument in `run.py` and `run_docker_evals.py` to `open_asr_librispeech_test_other`.
- Added `open_asr_librispeech_test_other` to the `AUDIO_EVAL_DATASETS` list in `eval_config.py`.
- Updated evaluation logic in `run_docker_evals.py` to include routing for the new Open-ASR task variant, improving task configuration clarity.

stisiTT committed:
Implement audio dataset transformation for evaluation and reporting workflows

- Added `apply_audio_dataset_transformation` function in `eval_config.py` to centralize audio dataset configuration logic for tasks named "librispeech".
- Updated `run_docker_evals.py` and `run_reports.py` to utilize the new transformation function, ensuring consistent handling of audio evaluation datasets across workflows.
- Enhanced argument parsing in `run_reports.py` to include `--audio-eval-dataset`, allowing for dynamic dataset selection during report generation.
- Improved logging to provide clearer insights into task updates and transformations applied during evaluation.

stisiTT committed:
Update evaluation configuration for Open-ASR tasks

- Renamed evaluation task from "librispeech" to "open_asr_librispeech_test_other" in `eval_config.py` to reflect the new dataset.
- Adjusted published scores and references in the evaluation tasks to align with the updated Open-ASR leaderboard.
- Enhanced `run_reports.py` to extract and format raw WER scores for reporting, improving clarity in evaluation results.

stisiTT commented:
### Accuracy Evaluations for distil-large-v3 on n300

| model           | device | task_name                       | accuracy_check | score | wer   | ratio_to_reference | gpu_reference_score | ratio_to_published | published_score                                                      |
|-----------------|--------|---------------------------------|----------------|-------|-------|--------------------|---------------------|--------------------|----------------------------------------------------------------------|
| distil-large-v3 | n300   | open_asr_librispeech_test_other | PASS ✅         | 94.32 | 5.68% | 0.98               | 96.19               | 0.99               | [94.81](https://huggingface.co/spaces/hf-audio/open_asr_leaderboard) |


### Accuracy Evaluations for whisper-large-v3 on n300

| model            | device | task_name                       | accuracy_check | score | wer   | ratio_to_reference | gpu_reference_score | ratio_to_published | published_score                                                      |
|------------------|--------|---------------------------------|----------------|-------|-------|--------------------|---------------------|--------------------|----------------------------------------------------------------------|
| whisper-large-v3 | n300   | open_asr_librispeech_test_other | PASS ✅         | 95.43 | 4.57% | 0.99               | 96.19               | 0.99               | [96.09](https://huggingface.co/spaces/hf-audio/open_asr_leaderboard) |

milank94 commented:
Daily:

- @bgoelTT to review and merge with [ben/whisper-evals](https://github.com/tenstorrent/tt-inference-server/tree/ben/whisper-evals)
- @stisiTT to rebase, post review

bgoelTT reviewed (approved):
Looks good Sam, I think once this gets rebased to support the Media Inference Server & the new `ModelSpec` we can remove a lot of the special Whisper handling

stisiTT committed:
Update Blackhole Default Implementations
